File: app/src/main/java/com/iug/coursehub/feature/user/my_courses_screen/my_course_list_screen/MyCoursesViewModel.java
package com.iug.coursehub.feature.user.my_courses_screen.my_course_list_screen;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.iug.coursehub.data.local.db.CoursesRepository;
import com.iug.coursehub.data.local.db.entity.Course;
import com.iug.coursehub.data.local.prefs.AppPrefs;

import java.util.List;

public class MyCoursesViewModel extends ViewModel {

    private MutableLiveData<List<Course>> _courses = new MutableLiveData<>();
    public LiveData<List<Course>> courses = _courses;

    public void getEnrolledCourses() {
        int userId = AppPrefs.getCurrentUserId();
        List<Integer> enrolledCourseIds = CoursesRepository.getEnrolledCourseIds(userId);
        List<Course> enrolledCourses = CoursesRepository.getAllCoursesOfIds(enrolledCourseIds);
        _courses.postValue(enrolledCourses);
    }
}

File: app/src/main/java/com/iug/coursehub/feature/user/my_courses_screen/my_course_list_screen/MyCoursesViewModel.kt
package com.iug.coursehub.feature.user.my_courses_screen.my_course_list_screen

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.iug.coursehub.data.local.db.CoursesRepository
import com.iug.coursehub.data.local.db.entity.Course
import com.iug.coursehub.data.local.prefs.AppPrefs
import kotlinx.coroutines.launch

class MyCoursesViewModel : ViewModel() {

    private val _courses = MutableLiveData<List<Course>>()
    val courses: LiveData<List<Course>> = _courses

    fun getEnrolledCourses() {
        viewModelScope.launch {
            val userId = AppPrefs.getCurrentUserId()
            val enrolledCourseIds = CoursesRepository.getEnrolledCourseIds(userId)
            val enrolledCourses = CoursesRepository.getAllCoursesOfIds(enrolledCourseIds)
            _courses.postValue(enrolledCourses)
        }
    }
}